docs/agent-persistency.md

```diff
@@ -106,78 +171,130 @@ suspend fun example(context: AIAgentContextBase) {
         nodeId = "current-node-id",
         lastInput = inputData
     )
-    
+
     // The checkpoint ID can be stored for later use
     val checkpointId = checkpoint.checkpointId
 }
 ```
 
+<!--- KNIT example-agent-persistency-05.kt -->
+
 ### Restoring from a checkpoint
 
 To restore the state of an agent from a specific checkpoint, follow the code sample below:
 
+<!--- INCLUDE
+import ai.koog.agents.core.agent.context.AIAgentContextBase
+import ai.koog.agents.snapshot.feature.persistency
+-->
+
 ```kotlin
 suspend fun example(context: AIAgentContextBase, checkpointId: String) {
     // Roll back to a specific checkpoint
     context.persistency().rollbackToCheckpoint(checkpointId, context)
-    
+
     // Or roll back to the latest checkpoint
     context.persistency().rollbackToLatestCheckpoint(context)
 }
 ```
 
+<!--- KNIT example-agent-persistency-06.kt -->
+
 ### Using extension functions
 
 The Agent Persistency feature provides convenient extension functions for working with checkpoints:
 
+<!--- INCLUDE
+import ai.koog.agents.core.agent.context.AIAgentContextBase
+import ai.koog.agents.snapshot.feature.persistency
+import ai.koog.agents.snapshot.feature.withPersistency
+
+const val inputData = "some-input-data"
+-->
+
 ```kotlin
 suspend fun example(context: AIAgentContextBase) {
     // Access the checkpoint feature
     val checkpointFeature = context.persistency()
-    
+
     // Or perform an action with the checkpoint feature
     context.withPersistency(context) { ctx ->
         // 'this' is the checkpoint feature
         createCheckpoint(ctx.id, ctx, "node-id", inputData)
     }
 }
 ```
+<!--- KNIT example-agent-persistency-07.kt -->
 
 ## Advanced usage
 
 ### Custom storage providers
 
 You can implement custom storage providers by implementing the `PersistencyStorageProvider` interface:
 
+<!--- INCLUDE
+import ai.koog.agents.snapshot.feature.AgentCheckpointData
+import ai.koog.agents.snapshot.providers.PersistencyStorageProvider
+-->
+
 ```kotlin
 class MyCustomStorageProvider : PersistencyStorageProvider {
     override suspend fun getCheckpoints(agentId: String): List<AgentCheckpointData> {
-        // Implementation
+        TODO("Implementation")
```

Comment:
To make code examples compilable, I need to put TODO(“Implementation”) instead of comment // Implementation.
Is it critical or maybe TODOs considered as a bad taste? WDYT?
I can just ignore this example, but maybe it's also fine like this?